File: spring_java_orm_sample/src/main/java/com/example/demo/jooq/dsl/BookGenreDsl.java
package com.example.demo.jooq.dsl;

import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import static com.example.demo.jooq.generated.Tables.BOOK_GENRE;

@Component
@Profile("jooq")
@RequiredArgsConstructor
public class BookGenreDsl {
    private final DSLContext dsl;

    public boolean exists(Long genreId) {
        return dsl.fetchExists(BOOK_GENRE, BOOK_GENRE.ID.eq(genreId));
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/jooq/dsl/BookGenreDsl.kt
package com.example.demo.jooq.dsl

import com.example.demo.jooq.generated.Tables
import org.jooq.DSLContext
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Component
@Profile("jooq")
class BookGenreDsl(private val dsl: DSLContext) {
    fun exists(genreId: Long?): Boolean {
        return dsl.fetchExists(Tables.BOOK_GENRE, Tables.BOOK_GENRE.ID.eq(genreId))
    }
}
